components/org.wso2.carbon.identity.organization.config.service/src/main/java/org/wso2/carbon/identity/organization/config/service/AttributeBasedOrganizationDiscoveryHandlerRegistry.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.organization.config.service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.wso2.carbon.identity.organization.config.service.constant.OrganizationConfigConstants.EMAIL_DOMAIN_BASED_SELF_SIGNUP_ENABLE;
+
+/**
+ * Attribute based organization discovery handler registry. This class holds the supported discovery attribute keys.
+ */
+public class AttributeBasedOrganizationDiscoveryHandlerRegistry {
+
+    private static final AttributeBasedOrganizationDiscoveryHandlerRegistry instance = new
```

Comment:
```suggestion
    private static final AttributeBasedOrganizationDiscoveryHandlerRegistry INSTANCE = new
```
since it is a final constant